sympy/stats/symbolic_probability.py

```diff
@@ -501,3 +501,32 @@ def _eval_rewrite_as_Integral(self, arg1, arg2, condition=None, **kwargs):
 
     def evaluate_integral(self):
         return self.rewrite(Integral).doit()
+
+class Moment(Expr):
+    """
+    Symbolic Moment class.
+    """
+    def __new__(cls, X, n, c=0, condition=None, **kwargs):
+        if not is_random(X):
+            return X
+        n = _sympify(n)
+        c = _sympify(c)
+        if condition is not None:
+            condition = _sympify(condition)
+            obj = Expr.__new__(cls, X, n, c, condition)
+        else:
+            obj = Expr.__new__(cls, X, n, c)
+        obj._condition = condition
```

Comment:
You don't need to store `condition` since it is one of the `args`. It should be `self.args[3]` although I would make an accessor property:
```
@property
def condition(self):
    return self.args[3]
```